judge/views/problem.py

```diff
@@ -200,9 +200,69 @@ def get_context_data(self, **kwargs):
                                           context['description'], 'problem')
         context['meta_description'] = self.object.summary or metadata[0]
         context['og_image'] = self.object.og_image or metadata[1]
+
+        context['can_vote'] = self.object.can_vote(user)
+        # The vote this user has already cast on this problem.
+        if context['can_vote']:
+            try:
+                context['vote'] = ProblemPointsVote.objects.get(voter=user.profile, problem=self.object)
+            except ObjectDoesNotExist:
+                context['vote'] = None
+        else:
+            context['vote'] = None
+
+        all_votes = list(self.object.problem_points_votes.order_by('points').values_list('points', flat=True))
+        context['has_votes'] = len(all_votes) > 0
+
+        # If the user is not currently in contest.
+        if not user.is_authenticated or user.profile.current_contest is None:
```

Comment:
```suggestion
        if not self.request.in_contest:
```
If we just want to check if the user is in contest, let's use this, which is more clear. Also, I think we can delete the comment above if we change to this since it's pretty obvious what we're checking for.